scripts/loader_tramp_term_generator.py

```diff
@@ -0,0 +1,2186 @@
+# Copyright (c) 2015-2022 LunarG, Inc.
+# Copyright (c) 2015-2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Mark Young <marky@lunarg.com>
+# Author: Mark Lobodzinski <mark@lunarg.com>
+
+import os,re,sys
+import xml.etree.ElementTree as etree
+from generator import *
+from collections import namedtuple
+from common_codegen import *
+
+# WSI extensions that should be ignored
+WSI_EXT_NAMES = [
+    'VK_KHR_android_surface',
+    'VK_KHR_display',
+    'VK_KHR_display_swapchain',
+    'VK_KHR_get_display_properties2',
+    'VK_KHR_get_surface_capabilities2',
+    'VK_KHR_surface',
+    'VK_KHR_swapchain',
+    'VK_KHR_wayland_surface',
+    'VK_KHR_win32_surface',
+    'VK_KHR_xcb_surface',
+    'VK_KHR_xlib_surface',
+    'VK_EXT_directfb_surface',
+    'VK_EXT_headless_surface',
+    'VK_EXT_metal_surface',
+    'VK_FUCHSIA_imagepipe_surface',
+    'VK_GGP_stream_descriptor_surface',
+    'VK_MVK_macos_surface',
+    'VK_MVK_ios_surface',
+    'VK_NN_vi_surface',
+    'VK_QNX_screen_surface',
+]
+
+# These are commands that should be ignored because we manually implement the
+# trampolines and terminators
+MANUALLY_IMPLEMENTED_COMMANDS = [
+    # 1.0 commands
+    'vkGetInstanceProcAddr',
+    'vkGetDeviceProcAddr',
+    'vkEnumerateInstanceExtensionProperties',
+    'vkEnumerateInstanceLayerProperties',
+    'vkEnumerateInstanceVersion',
+    'vkCreateInstance',
+    'vkDestroyInstance',
+    'vkEnumerateDeviceLayerProperties',
+    'vkCreateDevice',
+    'vkDestroyDevice',
+    'vkEnumeratePhysicalDevices',
+    # 1.1 commands
+    'vkEnumeratePhysicalDeviceGroups',
+    # Extension commands
+    # -- VK_EXT_debug_utils
+    'vkCreateDebugUtilsMessengerEXT',
+    'vkDestroyDebugUtilsMessengerEXT',
+    'vkSubmitDebugUtilsMessageEXT',
+    # -- VK_EXT_full_screen_exclusive
+    'vkGetPhysicalDeviceSurfacePresentModes2EXT'
+    'vkGetDeviceGroupSurfacePresentModes2EXT'
+]
+
+# Instance Extensions that need to be handled
+INST_EXTENSION_OVERRIDES = [
+    'VK_EXT_acquire_drm_display',
+    'VK_EXT_acquire_xlib_display',
+    'VK_EXT_debug_utils',
+    'VK_EXT_direct_mode_display',
+    'VK_EXT_display_surface_counter',
+    'VK_EXT_full_screen_exclusive',
+    'VK_NV_external_memory_capabilities',
+]
+
+MANUALLY_IMPLEMENTED_TERMINATORS = [
+    # 1.0 commands
+    'vkEnumerateInstanceExtensionProperties',
+    'vkEnumerateInstanceLayerProperties',
+    'vkEnumerateDeviceExtensionProperties',
```

Comment:
Because it's in the `MANUALLY_IMPLEMENTED_COMMANDS` because it already has both a trampoline and terminator manually defined.  The ones in this list only have manually terminators and need only to generate the trampoline.

Reply:
Ahhh my bad. That makes sense.